pkg/services/alerting/notifiers/sensugo.go

```diff
@@ -0,0 +1,195 @@
+package notifiers
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/alerting"
+)
+
+func init() {
+	alerting.RegisterNotifier(&alerting.NotifierPlugin{
+		Type:        "sensugo",
+		Name:        "Sensu Go",
+		Description: "Sends HTTP POST request to a Sensu Go API",
+		Heading:     "Sensu Go Settings",
+		Factory:     NewSensuGoNotifier,
+		Options: []alerting.NotifierOption{
+			{
+				Label:        "Backend URL",
+				Element:      alerting.ElementTypeInput,
+				InputType:    alerting.InputTypeText,
+				Placeholder:  "http://sensu-api.local:8080",
+				PropertyName: "url",
+				Required:     true,
+			},
+			{
+				Label:        "API Key",
+				Element:      alerting.ElementTypeInput,
+				InputType:    alerting.InputTypePassword,
+				Description:  "API Key to auth to Sensu Go backend",
+				PropertyName: "apikey",
+				Required:     true,
+			},
+			{
+				Label:        "Proxy entity name",
+				Element:      alerting.ElementTypeInput,
+				InputType:    alerting.InputTypeText,
+				Description:  "If empty, rule name will be used",
+				PropertyName: "entity",
+			},
+			{
+				Label:        "Check name",
+				Element:      alerting.ElementTypeInput,
+				InputType:    alerting.InputTypeText,
+				Description:  "If empty, rule id will be used",
+				PropertyName: "check",
+			},
+			{
+				Label:        "Handler",
+				Element:      alerting.ElementTypeInput,
+				InputType:    alerting.InputTypeText,
+				PropertyName: "handler",
+			},
+			{
+				Label:        "Namespace",
+				Element:      alerting.ElementTypeInput,
+				InputType:    alerting.InputTypeText,
+				Placeholder:  "default",
+				PropertyName: "namespace",
+			},
+		},
+	})
+}
+
+// NewSensuGoNotifier is the constructor for the Sensu Go Notifier.
+func NewSensuGoNotifier(model *models.AlertNotification) (alerting.Notifier, error) {
+	url := model.Settings.Get("url").MustString()
+	apikey := model.Settings.Get("apikey").MustString()
```

Comment:
```suggestion
	apikey := model.DecryptedValue("apikey", model.Settings.Get("apikey").MustString())
```
`model.DecryptedValue()` searches for the field in the SecureSettings and only if it's not there assigns the fallback string

This is required otherwise the following check:
https://github.com/grafana/grafana/blob/54f886e9118dffa23d5c34f4ab4ce4ee304c51f3/pkg/services/alerting/notifiers/sensugo.go#L78-L81
prevents from having a provisioning configuration where the api key is defined under secure_settings like:
```
notifiers:
  - name: provisioned-sensugo
    type: sensugo
    org_id: 1
    org_name: Main Org.
    uid: abcde
    is_default: false
    settings:
      url: "http://localhost:8080"
    secure_settings:
      apikey: "somekey"
```